python/kserve/kserve/protocol/rest/v1_endpoints.py

```diff
@@ -78,11 +79,12 @@ async def predict(self, model_name: str, request: Request) -> Union[Response, Di
                                                                 headers=headers)
         response, response_headers = self.dataplane.encode(model_name=model_name,
                                                            response=response,
-                                                           headers=headers, req_attributes=req_attributes)
+                                                           headers=response_headers, req_attributes=req_attributes)
 
-        if not isinstance(response, dict):
-            return Response(content=response, headers=response_headers)
-        return response
+        if isinstance(response, dict):
```

Comment:
should we use the same way that handles the response headers in v2 endpoints here? 
i.e. use the `raw_response`?

https://github.com/kserve/kserve/blob/49f662393c27d2a88ba32efd0ddeb38d7a5f3df3/python/kserve/kserve/protocol/rest/v2_endpoints.py#L147-L149

```
        response, response_headers = self.dataplane.encode(model_name=model_name,
                                                           response=response,
                                                           headers=response_headers, req_attributes={})

        if response_headers:
            raw_response.headers.update(response_headers)
        res = InferenceResponse.parse_obj(response)
        return res
  ```